sopel/modules/bugzilla.py

```diff
@@ -42,34 +40,35 @@ def configure(config):
 
 
 def setup(bot):
-    global regex
     bot.config.define_section('bugzilla', BugzillaSection)
 
-    if not bot.config.bugzilla.domains:
-        return
 
-    domains = '|'.join(bot.config.bugzilla.domains)
-    regex = re.compile((r'https?://(%s)'
-                        r'(/show_bug.cgi\?\S*?)'
-                        r'(id=\d+)')
-                       % domains)
-    bot.register_url_callback(regex, show_bug)
+def _bugzilla_loader(settings):
+    if not settings.bugzilla.domains:
+        raise plugins.exceptions.ImproperlyConfigured(
+            'Bugzilla URL callback requires '
+            '"bugzilla.domains" to be configured; check your config file.')
+
+    domain_pattern = '|'.join(
+        re.escape(domain)
+        for domain in settings.bugzilla.domains)
 
+    pattern = (
+        r'https?://(%s)'
+        r'(/show_bug.cgi\?\S*?)'
+        r'(id=\d+).*'
+    ) % domain_pattern
 
-def shutdown(bot):
-    bot.unregister_url_callback(regex, show_bug)
+    return [re.compile(pattern)]
 
 
-@rule(r'.*https?://(\S+?)'
-      r'(/show_bug.cgi\?\S*?)'
-      r'(id=\d+).*')
+@plugin.url_lazy(_bugzilla_loader)
 def show_bug(bot, trigger, match=None):
     """Show information about a Bugzilla bug."""
-    match = match or trigger
-    domain = match.group(1)
+    domain = trigger.group(1)
     if domain not in bot.config.bugzilla.domains:
         return
```

Comment:
Why do we still need this if the whole thing's using URL callbacks now? It'll never match unless the domain was in the config list when `_bugzilla_loader()` was called by the rule setup.

Reply:
Lol at me.